internal/services/appconfiguration/app_configuration_key_resource.go

```diff
@@ -0,0 +1,274 @@
+	return map[string]*schema.Schema{}
+}
+
+func (k KeyResource) ModelObject() interface{} {
+	return nil
+}
+
+func (k KeyResource) ResourceType() string {
+	return "azurerm_app_configuration_key"
+}
+
+func (k KeyResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model KeyResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			client, err := metadata.Client.AppConfiguration.DataPlaneClient(ctx, model.ConfigurationStoreId)
+			if err != nil {
+				return err
+			}
+
+			appCfgKeyResourceID := parse.AppConfigurationKeyId{
+				ConfigurationStoreId: model.ConfigurationStoreId,
+				Key:                  model.Key,
+				Label:                model.Label,
+			}
+
+			kv, err := client.GetKeyValues(ctx, model.Key, model.Label, "", "", []string{})
+			if err != nil {
+				return fmt.Errorf("while checking for key's %q existence: %+v", model.Key, err)
+			}
+			keysFound := kv.Values()
+			if len(keysFound) > 0 {
+				return tf.ImportAsExistsError(k.ResourceType(), appCfgKeyResourceID.ID())
+			}
+
+			entity := appconfiguration.KeyValue{
+				Key:         utils.String(model.Key),
+				Label:       utils.String(model.Label),
+				ContentType: utils.String(model.ContentType),
+				Value:       utils.String(model.Value),
+				Tags:        tags.Expand(model.Tags),
+			}
+
+			_, err = client.PutKeyValue(ctx, model.Key, model.Label, &entity, "", "")
+			if err != nil {
+				return err
+			}
+
+			if model.Locked {
+				_, err = client.PutLock(ctx, model.Key, model.Label, "", "")
+				if err != nil {
+					return fmt.Errorf("while locking key/label pair %s/%s: %+v", model.Key, model.Label, err)
+				}
+			}
+
+			metadata.SetID(appCfgKeyResourceID)
+			return nil
+		},
+		Timeout: 30 * time.Minute,
+	}
+}
+
+func (k KeyResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			resourceID, err := parse.AppConfigurationKeyID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("while parsing resource ID: %+v", err)
+			}
+
+			client, err := metadata.Client.AppConfiguration.DataPlaneClient(ctx, resourceID.ConfigurationStoreId)
+			if err != nil {
+				return err
+			}
+
+			res, err := client.GetKeyValues(ctx, resourceID.Key, resourceID.Label, "", "", []string{})
+			if err != nil {
+				if !utils.ResponseWasNotFound(res.Response().Response) {
+					return metadata.MarkAsGone(resourceID)
+				}
+				return fmt.Errorf("while checking for key's %q existence: %+v", resourceID.Key, err)
+			}
+
+			if len(res.Values()) > 1 {
+				return fmt.Errorf("unexpected API response. More than one value returned for Key/Label pair %s/%s", resourceID.Key, resourceID.Label)
+			}
+
+			kv := res.Values()[0]
+			var locked bool
+			if kv.Locked != nil {
+				locked = *kv.Locked
+			}
```

Comment:
just a thought but you technically don't need the `locked` variable here you could move this below and direct-assign it (since `model.Locked` would have a default of false?)